cocotb/decorators.py

```diff
@@ -99,10 +99,6 @@ def __init__(self, inst, parent):
             self._coro = inst
         self._started = False
         self._callbacks = []
-        self._parent = parent
-        self.__doc__ = parent._func.__doc__
-        self.module = parent._func.__module__
-        self.funcname = parent._func.__name__
         self._outcome = None
 
         if not hasattr(self._coro, "send"):
```

Comment:
Like in my PR this should be moved to earlier in the method and check `inspect.isgenerator` instead of checking for only `send` for non-async-coroutine cases.